homeassistant/components/schluter/climate.py

```diff
@@ -0,0 +1,161 @@
+"""Support for Schluter thermostats."""
+from datetime import timedelta
+import logging
+
+from requests import RequestException
+import voluptuous as vol
+
+from homeassistant.components.climate import PLATFORM_SCHEMA, ClimateDevice
+from homeassistant.components.climate.const import (
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, CONF_SCAN_INTERVAL
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from . import DATA_SCHLUTER, DOMAIN as SCHLUTER_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = timedelta(seconds=5)
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Optional(CONF_SCAN_INTERVAL): vol.All(vol.Coerce(int), vol.Range(min=1))}
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Schluter thermostats."""
+    session_id = hass.data[DATA_SCHLUTER].session_id
+    api = hass.data[DATA_SCHLUTER].api
+    temp_unit = hass.config.units.temperature_unit
+
+    async def async_update_data():
+        try:
+            return api.get_thermostats(session_id) or []
+        except RequestException as err:
+            raise UpdateFailed(f"Error communicating with Schluter API: {err}")
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="schluter",
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=10),
+    )
+
+    await coordinator.async_refresh()
+
+    async_add_entities(
+        SchluterThermostat(coordinator, idx, temp_unit, api, session_id)
+        for idx, ent in enumerate(coordinator.data)
+    )
+
+
+class SchluterThermostat(ClimateDevice):
+    """Representation of a Schluter thermostat."""
+
+    def __init__(self, coordinator, idx, temp_unit, api, session_id):
+        """Initialize the thermostat."""
+        self._unit = temp_unit
+        self.coordinator = coordinator
+        self.idx = idx
+        self._api = api
+        self._session_id = session_id
+        self._support_flags = SUPPORT_TARGET_TEMPERATURE
+
+    @property
+    def should_poll(self):
+        """Return if platform should poll."""
+        return True
```

Comment:
Polling is not needed when using the coordinator. The entity should subscribe to changes from the data coordinator:

```py
    async def async_added_to_hass(self):
        """When entity is added to hass."""
        self.coordinator.async_add_listener(
            self.async_write_ha_state
        )

    async def async_will_remove_from_hass(self):
        """When entity will be removed from hass."""
        self.coordinator.async_remove_listener(
            self.async_write_ha_state
        )
```

See coordinator developer documentation for more information and more extensive example